src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Bindings/WatchableCloudBlobStream.cs

```diff
@@ -8,24 +8,41 @@
 using Microsoft.Azure.WebJobs.Host.Bindings;
 using Microsoft.Azure.WebJobs.Host.Protocols;
 using Microsoft.Azure.Storage.Blob;
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+using System.Collections.Generic;
 
 namespace Microsoft.Azure.WebJobs.Host.Blobs.Bindings
 {
     internal class WatchableCloudBlobStream : DelegatingCloudBlobStream, IWatcher
     {
+        private readonly ILogger _logger;
+        private readonly ICloudBlob _blob;
```

Comment:
`_blob` and `_timeWrite` are only used for logging. Instead of creating `private` fields for these two add a private property `JObject _blobAccessLog`  and start populating `_blobAccessLog` in constructor:

```
_blobAccessLog["BlobName"] = blob.Name 
```

and so on...